js-weather.js

```diff
@@ -0,0 +1,81 @@
+document.addEventListener('DOMContentLoaded', () => {
+  const targetCity = document.getElementById('city');
+  const appId = '4b5774e9f3d2a07b84f0f2f88e486224';
+
+  const WeatherGetAsync = () => {
+    const cityName = targetCity.value;
+    const requestUrl = 'https://api.openweathermap.org/data/2.5/weather?APPID=' + appId + '&lang=ja&units=metric&q=' + cityName + ',jp;';
+
+    const result = document.getElementById('result');
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', requestUrl);
+    xhr.send();
+
+    xhr.addEventListener('loadstart', () => {
+      result.textContent = '通信中...';
+    }, false);
+
+    xhr.addEventListener('load', () => {
+      showTodaysWeather(xhr.responseText);
+    }, false);
+
+    xhr.addEventListener('error', () => {
+      result.textContent = 'サーバーエラーが発生しました。';
+    }, false);
+  }
+    
+  WeatherGetAsync();
+  targetCity.addEventListener('change', WeatherGetAsync);
+
+}, false);
+
+function showTodaysWeather(response){
```

Comment:
```suggestion
const showTodaysWeather = (response) => {
```

アロー関数にしましょう。アロー関数にするにあたり、関数を最初にコールする個所より上で定義しましょう。関数定義をファイルの上の方にまとめるとよいという事です。